logstash-core/lib/logstash/logging/slow_logger.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging/slow_logger/freq_items"
+
+module LogStash
+  module Logging
+
+    class NullLogger
+      def log(threshold, time, data={})
+      end
+      alias_method :warn, :log
+    end
+
+    class SlowLogger
+
+      attr_reader :logger, :freq_items
+
+      def initialize(name="loggers.slow", params={})
+        @logger     = LogStash::Logging::Logger.new(name)
+        @freq_items = LogStash::Logging::Util::FreqItems.new
+      end
+
+      def log(threshold, time, data={})
```

Comment:
Adding the event is totally possible now, but looks good to me to make it a requirement. At the beginning I thought not to force it, so left the data hash there to let users really add whatever they prefer. 

Not completely sure the best way would it be to output base on this setting, might be only output the _context_ when running in debug more? what do you think?